bumble/utils.py

```diff
@@ -59,6 +59,32 @@ def deregister(self, emitter):
     return cls
 
 
+# -----------------------------------------------------------------------------
+class EventEmitter(BaseEventEmitter):
```

Comment:
I like how this class simplifies a lot of the multi-event waiting we need to do in the `Device` class. Since async tasks, awaitables and cancellation can be tricky sometimes, it would be great to add some tests to make sure that this class behaves as expected in different situations, and avoid difficult-to-trace regressions if we make modifications to it later.

Reply:
I added a "very" basic test, this might not be enough